File: LeetCode Solutions/May 2020 LeetCoding Challenges/Maximum Erasure Value.py
class Solution:
    def maximumUniqueSubarray(self, nums: List[int]) -> int:
        nmap, Sum, maximumScore, left = [0] * 10001, 0, 0, 0
        for right in nums:
            nmap[right] += 1
            Sum += right
            while nmap[right] > 1:
                nmap[nums[left]] -= 1
                Sum -= nums[left]
                left += 1
            maximumScore = max(maximumScore, Sum)
        return maximumScore

File: LeetCode Solutions/LeetCoding Challenge/May 2020 LeetCoding Challenge/Maximum Erasure Value.java
class Solution {
    public int maximumUniqueSubarray(int[] nums) {
        short[] nmap = new short[10001];
        int Sum = 0, maximumScore = 0;
        for (int left = 0, right = 0; right < nums.length; right++) {
            nmap[nums[right]]++;
            Sum += nums[right];
            while (nmap[nums[right]] > 1) {
                nmap[nums[left]]--;
                Sum -= nums[left++];
            }
            maximumScore = Math.max(maximumScore, Sum);
        }
        return maximumScore;
    }
}